src/pytorch_lightning/loggers/logger.py

```diff
@@ -13,29 +13,35 @@
 # limitations under the License.
 """Abstract base class used to build new loggers."""
 
-import argparse
+
 import functools
 import operator
+import typing as t
 from abc import ABC, abstractmethod
 from argparse import Namespace
+from collections import defaultdict
 from functools import wraps
-from typing import Any, Callable, Dict, Iterable, List, Mapping, Optional, Sequence, Union
+from typing import Any, Callable, Dict, Generator, Iterable, List, Mapping, Optional, Sequence, Union
 from weakref import ReferenceType
 
 import numpy as np
+from torch import Tensor
 
 import pytorch_lightning as pl
 from pytorch_lightning.callbacks import Checkpoint
 from pytorch_lightning.utilities.rank_zero import rank_zero_deprecation, rank_zero_only
 
+if t.TYPE_CHECKING:
+    from pytorch_lightning.loggers.csv_logs import ExperimentWriter
+
 
 def rank_zero_experiment(fn: Callable) -> Callable:
     """Returns the real experiment on rank 0 and otherwise the DummyExperiment."""
 
     @wraps(fn)
-    def experiment(self):
+    def experiment(self: Callable) -> Union[ExperimentWriter, DummyExperiment]:
```

Comment:
We shouldn't annotate self. self has always the pseudo type "Self"

Reply:
When I remove that annotation, mypy complains:

`pytorch_lightning/loggers/logger.py:42: error: Function is missing a type annotation for one or more arguments`.

Rather than annotate Self, should I comment with `# type: ignore[no-untyped-def]` to exclude the error?